File: dotnet/Examples/BasicUsage.cs
using System;
using System.Threading.Tasks;
using CyberAPI;

namespace CyberAPI.Examples
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Get API key from environment or use demo
            var apiKey = Environment.GetEnvironmentVariable("CYBERAPI_KEY") ?? "demo";
            
            Console.WriteLine("🔍 CyberAPI .NET SDK - Basic Usage Examples\n");
            
            using var client = new CyberAPIClient(apiKey);
            
            // Example 1: Domain Check
            await Example1_DomainCheck(client);
            
            // Example 2: Quick Malicious Check
            await Example2_MaliciousCheck(client);
            
            // Example 3: Get Signals
            await Example3_GetSignals(client);
            
            // Example 4: IP Geolocation
            await Example4_IpGeolocation(client);
            
            // Example 5: Error Handling
            await Example5_ErrorHandling(client);
            
            Console.WriteLine("\n" + new string('=', 50));
            Console.WriteLine("✨ Examples completed!");
            Console.WriteLine(new string('=', 50));
        }
        
        static async Task Example1_DomainCheck(CyberAPIClient client)
        {
            Console.WriteLine(new string('=', 50));
            Console.WriteLine("Example 1: Domain Check");
            Console.WriteLine(new string('=', 50));
            
            try
            {
                var result = await client.CheckAsync(domain: "example.com");
                
                Console.WriteLine($"Domain: {result.Domain}");
                Console.WriteLine($"Risk Score: {result.RiskScore}/100");
                Console.WriteLine($"Verdict: {result.Verdict}");
                Console.WriteLine($"Cached: {result.Cached}");
                Console.WriteLine($"Process Time: {result.ProcessTimeMs}ms");
                
                if (result.Signals != null && result.Signals.Count > 0)
                {
                    Console.WriteLine("\nThreat Signals:");
                    foreach (var signal in result.Signals)
                    {
                        Console.WriteLine($"  ⚠️  {signal}");
                    }
                }
                else
                {
                    Console.WriteLine("\n✅ No threat signals detected");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"❌ Error: {ex.Message}");
            }
        }
        
        static async Task Example2_MaliciousCheck(CyberAPIClient client)
        {
            Console.WriteLine("\n" + new string('=', 50));
            Console.WriteLine("Example 2: Quick Malicious Check");
            Console.WriteLine(new string('=', 50));
            
            try
            {
                var domain = "example.com";
                var isMalicious = await client.IsMaliciousAsync(domain: domain);
                
                if (isMalicious)
                {
                    Console.WriteLine($"⚠️  {domain} is MALICIOUS!");
                }
                else
                {
                    Console.WriteLine($"✅ {domain} appears safe");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"❌ Error: {ex.Message}");
            }
        }
        
        static async Task Example3_GetSignals(CyberAPIClient client)
        {
            Console.WriteLine("\n" + new string('=', 50));
            Console.WriteLine("Example 3: Extract Signals");
            Console.WriteLine(new string('=', 50));
            
            try
            {
                var signals = await client.GetSignalsAsync(domain: "example.com");
                
                Console.WriteLine($"Detected {signals.Count} signals:");
                if (signals.Count > 0)
                {
                    foreach (var signal in signals)
                    {
                        Console.WriteLine($"  - {signal}");
                    }
                }
                else
                {
                    Console.WriteLine("  (none)");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"❌ Error: {ex.Message}");
            }
        }
        
        static async Task Example4_IpGeolocation(CyberAPIClient client)
        {
            Console.WriteLine("\n" + new string('=', 50));
            Console.WriteLine("Example 4: IP Geolocation");
            Console.WriteLine(new string('=', 50));
            
            try
            {
                var result = await client.CheckAsync(ip: "8.8.8.8");
                var geo = result.GeoLocation;
                
                Console.WriteLine($"IP: {geo?.Ip ?? "N/A"}");
                Console.WriteLine($"Country: {geo?.Country ?? "Unknown"}");
                Console.WriteLine($"City: {geo?.City ?? "Unknown"}");
                Console.WriteLine($"ISP: {geo?.Isp ?? "Unknown"}");
                Console.WriteLine($"Risk Score: {result.RiskScore}/100");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"❌ Error: {ex.Message}");
            }
        }
        
        static async Task Example5_ErrorHandling(CyberAPIClient client)
        {
            Console.WriteLine("\n" + new string('=', 50));
            Console.WriteLine("Example 5: Error Handling");
            Console.WriteLine(new string('=', 50));
            
            try
            {
                // This will throw an exception - missing both domain and ip
                await client.CheckAsync();
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine($"✅ Caught expected error: {ex.Message}");
            }
            catch (CyberAPIException ex)
            {
                Console.WriteLine($"API Error: {ex.Message} (HTTP {ex.StatusCode})");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Unexpected error: {ex.Message}");
            }
        }
    }
}

File: ruby/examples/basic_usage.rb
#!/usr/bin/env ruby

require_relative '../cyberapi'

# Example: Basic usage of CyberAPI Ruby SDK

# Get API key from environment or use demo
api_key = ENV['CYBERAPI_KEY'] || 'demo'

# Initialize client
client = CyberAPI::Client.new(api_key: api_key)

puts "🔍 CyberAPI Ruby SDK - Basic Usage Example\n\n"

# Example 1: Check a domain
puts "=" * 50
puts "Example 1: Domain Check"
puts "=" * 50

begin
  result = client.check(domain: "example.com")
  
  puts "Domain: #{result['domain']}"
  puts "Risk Score: #{result['risk_score']}/100"
  puts "Verdict: #{result['verdict']}"
  puts "Cached: #{result['cached'] ? 'Yes' : 'No'}"
  puts "Process Time: #{result['process_time_ms']}ms"
  
  if result['signals'].any?
    puts "\nThreat Signals:"
    result['signals'].each { |signal| puts "  ⚠️  #{signal}" }
  else
    puts "\n✅ No threat signals detected"
  end
rescue CyberAPI::Error => e
  puts "❌ Error: #{e.message}"
end

# Example 2: Quick malicious check
puts "\n" + "=" * 50
puts "Example 2: Quick Malicious Check"
puts "=" * 50

begin
  domain = "example.com"
  is_malicious = client.malicious?(domain: domain)
  
  if is_malicious
    puts "⚠️  #{domain} is MALICIOUS!"
  else
    puts "✅ #{domain} appears safe"
  end
rescue CyberAPI::Error => e
  puts "❌ Error: #{e.message}"
end

# Example 3: Get signals only
puts "\n" + "=" * 50
puts "Example 3: Extract Signals"
puts "=" * 50

begin
  signals = client.signals(domain: "example.com")
  
  puts "Detected #{signals.length} signals:"
  if signals.any?
    signals.each { |signal| puts "  - #{signal}" }
  else
    puts "  (none)"
  end
rescue CyberAPI::Error => e
  puts "❌ Error: #{e.message}"
end

# Example 4: IP check with geolocation
puts "\n" + "=" * 50
puts "Example 4: IP Geolocation"
puts "=" * 50

begin
  result = client.check(ip: "8.8.8.8")
  geo = result['geo_location']
  
  puts "IP: #{geo['ip']}"
  puts "Country: #{geo['country'] || 'Unknown'}"
  puts "City: #{geo['city'] || 'Unknown'}"
  puts "ISP: #{geo['isp'] || 'Unknown'}"
  puts "Risk Score: #{result['risk_score']}/100"
rescue CyberAPI::Error => e
  puts "❌ Error: #{e.message}"
end

puts "\n" + "=" * 50
puts "✨ Examples completed!"
puts "=" * 50
